### PR TITLE
AVRO-1907 - Add logging to logical type handling in java library

### PR DESCRIPTION
**Pull request by gszadovszky**
Didn't find other parts related to logical type handling where we would swallow an exception and therefore, some logging is required. 
Please, review and comment any idea if I should look into something more deeply.

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java [hunk 1]
```diff
@@ -83,6 +88,9 @@ private static LogicalType fromSchemaImpl(Schema schema, boolean throwErrors) {
     } catch (RuntimeException e) {
       if (throwErrors) {
         throw e;
+      } else {
+        LOGGER.warn("Ignoring invalid logical type for name: {}", typeName);
+        LOGGER.debug("Invalid logical type found", e);
```
Comment (zicl) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Optional: I would suggest moving the `LOGGER.debug` line before the `if`.

Comment (gszadovszky) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Make sense. Updated. 
Thanks for the review.

File: lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java [hunk 2]
```diff
@@ -81,8 +86,11 @@ private static LogicalType fromSchemaImpl(Schema schema, boolean throwErrors) {
         logicalType.validate(schema);
       }
     } catch (RuntimeException e) {
+      LOGGER.debug("Invalid logical type found", e);
       if (throwErrors) {
         throw e;
+      } else {
+        LOGGER.warn("Ignoring invalid logical type for name: {}", typeName);
```
Comment (zicl) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Ah sorry, one more minor issue. All the lines below this execute if throwErrors is false, but only the `LOGGER.warn` line is the else branch, which is a bit confusing/inconsistent. Could you move it out of the else, immediately after the if block like the existing code?

Comment (zicl) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Minor clarification: By "all the lines" I meant the single `logicalType = null;` line. :)

Comment (gszadovszky) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Good point. However the whole else branch is useless. Removed.

File: lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java [hunk 3]
```diff
@@ -21,8 +21,13 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class LogicalTypes {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(LogicalTypes.class);
```
Comment (tomwhite) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Please change to `LOG` for consistency with the rest of the codebase.

Comment (gszadovszky) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java]: Sorry, @tomwhite, I've missed your comment. Modified the name.

**Comment by rdblue** [comment]
Merged. Thanks @gszadovszky! Please close this PR.

**Pull request closed without merging**